tools/cli/src/cli/lib.rs

```diff
@@ -139,3 +179,571 @@ pub(crate) fn password_prompt(prompt: &str) -> Option<String> {
     }
     None
 }
+
+// TODO: this regex is also used in the webui (https://github.com/kanidm/kanidm/blob/003234c2d0a52146683628156e2a106bf61fe9f4/server/web_ui/src/views/identityverification.rs#L60) should we move it to proto?
```

Comment:
My only issue with moving it to proto is that it kinda felt misplaced, like everything inside proto is some sort of message, this would be quite different from everything else.

Reply:
You could probably dump the regex entirely, and just do a length check since you're already validating it's a number with the u32 parse

Reply:
yep that actually makes lots of sense. Initially I didn't do it because I thought "what if in the future we want to parse something that's not a number and we'll have to write the regex from scratch", but actually if that ever happens then we'll write the regex as needed.